test/run

```diff
@@ -969,8 +969,46 @@ testGodepPackageSpec() {
   assertCompiledBinaryExists
 }
 
+testGoCacheGoVersionGreaterThanOrEqualTo110() {
+  local cache_dir="${CACHE_DIR}/go1.10.1/build-cache"
+
+  fixture "dep-go-1-10"
+
+  assertDetected
+  assertDirDoesNotExist $cache_dir
+
+  compile
+  assertCaptured "Installing go1.10.1"
+  assertCaptured "Fetching dep"
+  assertCaptured "Fetching any unsaved dependencies (dep ensure)"
+  assertCaptured "Running: go install -v -tags heroku ."
+  assertCaptured "github.com/heroku/fixture"
+  assertCapturedSuccess
+  assertCompiledBinaryExists
+  assertDirExists $cache_dir
```

Comment:
Do we want to assert that something specific made it's way into the dir as well?